x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/optimizer/TomlFoldTests.java

```diff
@@ -0,0 +1,118 @@
+        filteredSpecs.removeAll(unsupportedSpecs);
+        return asArray(filteredSpecs);
+    }
+
+    public static List<Object[]> asArray(Collection<EqlFoldSpec> specs) {
+        AtomicInteger counter = new AtomicInteger();
+        return specs.stream().map(spec -> new Object[] {
+            counter.incrementAndGet(), spec
+        }).collect(toList());
+    }
+
+    private static final EqlConfiguration caseSensitiveConfig = config(true);
+    private static final EqlConfiguration caseInsensitiveConfig = config(false);
+    private static EqlParser parser = new EqlParser();
+    private static final EqlFunctionRegistry functionRegistry = new EqlFunctionRegistry();
+
+    private final int num;
+    private final EqlFoldSpec spec;
+
+    private static EqlConfiguration config(boolean isCaseSensitive) {
+        return new EqlConfiguration(new String[]{"none"},
+            org.elasticsearch.xpack.ql.util.DateUtils.UTC, "nobody", "cluster",
+            null, TimeValue.timeValueSeconds(30), -1, false, isCaseSensitive, "",
+            new TaskId("foobarbaz", 1234567890), () -> false);
+    }
+
+    public TomlFoldTests(int num, EqlFoldSpec spec) {
+        this.num = num;
+        this.spec = spec;
+    }
+
+    private void testWithConfig(EqlConfiguration testConfig, EqlFoldSpec spec) {
+        Expression expr = parser.createExpression(spec.expression());
+
+        // resolve all function calls
+        expr = expr.transformUp(e -> {
+            if (e instanceof UnresolvedFunction) {
+                UnresolvedFunction uf = (UnresolvedFunction) e;
+
+                if (uf.analyzed()) {
+                    return uf;
+                }
+
+                String name = uf.name();
+
+                if (uf.childrenResolved() == false) {
+                    return uf;
+                }
+
+                String functionName = functionRegistry.resolveAlias(name);
+                if (functionRegistry.functionExists(functionName) == false) {
+                    return uf.missing(functionName, functionRegistry.listFunctions());
+                }
+                FunctionDefinition def = functionRegistry.resolveFunction(functionName);
+                Function f = uf.buildResolved(testConfig, def);
+                return f;
+            }
+            return e;
+        });
+
+        assertTrue(expr.foldable());
+        Object folded = expr.fold();
+
+        // upgrade to a long, because the parser typically downgrades Long -> Integer when possible
+        if (folded instanceof Integer) {
+            folded  = ((Integer) folded).longValue();
+        }
+
+        assertEquals(spec.expected(), folded);
+    }
+
+    public void test() {
+        if (this.spec.supportsCaseSensitive()) {
+            testWithConfig(caseSensitiveConfig, spec);
+        }
+
+        if (this.spec.supportsCaseInsensitive()) {
+            testWithConfig(caseInsensitiveConfig, spec);
+        }
+    }
```

Comment:
This is better as a if/else (goes back to the it's either case sensitive or insensitive) - otherwise we might have tests that pass without anything being tested.